File: flowcatalyst-typescript/apps/platform/src/domain/auth-config/auth-config-type.ts
/**
 * Auth Config Type
 *
 * Type of authentication configuration, determining user scope and access.
 */

/**
 * Auth config types - determines user access scope.
 * - ANCHOR: Platform-wide access, users get ANCHOR scope (all clients)
 * - PARTNER: Partner access, users get PARTNER scope (granted clients only)
 * - CLIENT: Client-specific access, users get CLIENT scope (primary + additional clients)
 */
export type AuthConfigType = 'ANCHOR' | 'PARTNER' | 'CLIENT';

export const AuthConfigType = {
	ANCHOR: 'ANCHOR' as const,
	PARTNER: 'PARTNER' as const,
	CLIENT: 'CLIENT' as const,
} as const;

File: core/flowcatalyst-platform/src/main/java/tech/flowcatalyst/platform/client/AuthConfigType.java
package tech.flowcatalyst.platform.client;

import tech.flowcatalyst.platform.principal.UserScope;

/**
 * Type of authentication configuration, determining user scope and access.
 *
 * This is distinct from AuthProvider (INTERNAL vs OIDC) which determines
 * the authentication mechanism. AuthConfigType determines the access scope
 * of users authenticating through this configuration.
 */
public enum AuthConfigType {
    /**
     * Platform-wide/anchor configuration.
     * Users authenticating through this config get ANCHOR scope (access all clients).
     * Cannot have any client associations.
     */
    ANCHOR,

    /**
     * Partner configuration.
     * Users authenticating through this config get PARTNER scope.
     * Has explicitly granted client IDs stored on the config.
     * Users can only access the granted clients.
     */
    PARTNER,

    /**
     * Client-specific configuration.
     * Users authenticating through this config get CLIENT scope.
     * Must have a primary client, optionally with additional client exceptions.
     */
    CLIENT;

    /**
     * Convert this auth config type to the corresponding user scope.
     */
    public UserScope toUserScope() {
        return switch (this) {
            case ANCHOR -> UserScope.ANCHOR;
            case PARTNER -> UserScope.PARTNER;
            case CLIENT -> UserScope.CLIENT;
        };
    }
}
